### PR TITLE
fix: filter column is not result column

### PR DESCRIPTION
**Pull request by tpluscode**
The `Dimensions` class iterates all columns and marks them with `isResult` and `isFilter`

I think the two flags are mutually exclusive but a filter column would have been marked as "result" causing issues downstream when we try to generate CSV from a view

### CODE REVIEW

**Review by cristianvasquez — APPROVED** [review]
Looks ok to me, thanks!

**Comment by tpluscode** [comment]
It cannot look ok if there are tests failing :D

**Comment by tpluscode** [comment]
There, I had to change that logic to take `view:projection` into account. If present, only those columns are marked as `isResult` because only those will be part of the SELECT clause

**Comment by cristianvasquez** [comment]
What is the modern way to review pull requests with an instantiated server?

File: test/ViewQuery/Dimensions.test.js [hunk 1]
```diff
@@ -78,22 +61,71 @@ ex:view a view:View ;
...
+                ] ;
+            ] .
+      `.node(ns.ex.view)
 
       const dimensions = new Dimensions({ view, variable: () => {} })
 
       strictEqual('BlankNode', dimensions.array[0].property.termType)
     })
+
+  it('does not set isResult flag when dimension is not used in projection', () => {
```
Comment (cristianvasquez) [inline test/ViewQuery/Dimensions.test.js:78]: This one is testing for 'set isResult flag when dimension is used in projection'

Falsifiable by the followind data

```js
    const view = parse`
      ${ns.ex.view} a ${ns.view.View} ;
        ${ns.view.projection} [
          ${ns.view.columns} ( _:dimension _:excluded ) ;
        ] ;
        ${ns.view.dimension} _:dimension, _:excluded .
        
      _:dimension ${ns.view.from} [
        ${ns.view.path} ${ns.ex.some} 
      ] .
      
      _:excluded ${ns.view.from} [
        ${ns.view.path} ${ns.ex.notThat} 
      ] .
      ```

Comment (tpluscode) [reply test/ViewQuery/Dimensions.test.js:78]: Good catch. This is why I prefer TypeScript. The problem was

```
-       ? projectedColumns.has(dimension)
+       ? projectedColumns.has(dimension.term)
```

Which would have been caught by the type system

**Comment by cristianvasquez** [comment]
Needs standard  fix

**Comment by tpluscode** [comment]
> Wondering how one can review branches without having to setup all the environments, and run/modify tests there.

Gitpod, potentially

**Pull request merged**